MapboxNavigation/NavigationMapView.swift

```diff
@@ -719,6 +722,55 @@ open class NavigationMapView: MGLMapView {
         }
     }
     
+    public func showVoiceInstructionsOnMap(routeProgress: RouteProgress) {
+        guard let style = style else {
+            return
+        }
+        
+        let instructions = Array(routeProgress.route.legs.flatMap {
+            return $0.steps.flatMap { (step: RouteStep) in
+                return step.instructionsSpokenAlongStep.flatMap {
+                    return $0.flatMap { (instruction: SpokenInstruction) in
+                        return (Polyline(step.coordinates!).coordinateFromStart(distance: instruction.distanceAlongStep)!, instruction.text)
+                    }
+                }
+            }
+        }.joined())
+        var features = [MGLPointFeature]()
+        
+        for instruction in instructions {
+            let feature = MGLPointFeature()
+            feature.coordinate = instruction.0
+            feature.attributes = [ "instruction": instruction.1 ]
+            features.append(feature)
+        }
+        
+        let instructionPointSource = MGLShapeCollectionFeature(shapes: features)
+        
+        if let instructionSource = style.source(withIdentifier: instructionSource) as? MGLShapeSource {
+            instructionSource.shape = instructionPointSource
+        } else {
+            let sourceShape = MGLShapeSource(identifier: instructionSource, shape: instructionPointSource, options: nil)
+            style.addSource(sourceShape)
+            
+            let symbol = MGLSymbolStyleLayer(identifier: instructionLabel, source: sourceShape)
+            symbol.text = MGLStyleValue(rawValue: "{instruction}")
+            symbol.textFontSize = MGLStyleValue(rawValue: 14)
+            symbol.textHaloWidth = MGLStyleValue(rawValue: 0.25)
```

Comment:
Is this enough of a halo? I don’t see it in the screenshot in https://github.com/mapbox/mapbox-navigation-ios/pull/727/#issue-266279342. The halo would be useful against the night style.